pkg/metrics/middleware.go

```diff
@@ -24,13 +28,30 @@ import (
 // around API calls.
 // It records the timestamp of an API call in the provided gauge.
 type APIEventTSHelper struct {
-	Next    http.Handler
-	TSGauge prometheus.Gauge
+	Next      http.Handler
+	TSGauge   prometheus.Gauge
+	Histogram *prometheus.HistogramVec
+}
+
+type responderWrapper struct {
+	http.ResponseWriter
+	code int
+}
+
+func (rw *responderWrapper) WriteHeader(code int) {
+	rw.code = code
+	rw.ResponseWriter.WriteHeader(code)
 }
 
 // ServeHTTP implements the http.Handler interface. It records the timestamp
 // this API call began at, then chains to the next handler.
 func (m *APIEventTSHelper) ServeHTTP(r http.ResponseWriter, req *http.Request) {
 	m.TSGauge.SetToCurrentTime()
-	m.Next.ServeHTTP(r, req)
+	duration := spanstat.Start()
+	rw := &responderWrapper{ResponseWriter: r}
+	m.Next.ServeHTTP(rw, req)
+	if req != nil && req.URL != nil && req.URL.Path != "" {
+		took := float64(duration.End(true).Total().Nanoseconds()) / float64(time.Second)
+		m.Histogram.WithLabelValues(req.URL.Path, req.Method, strconv.Itoa(rw.code)).Observe(took)
```

Comment:
I'm a bit against of this Path in the labels, the Prometheus labels will grow  a  lot and users can have issues with multiple labels. I'm not sure if this is the correct behaviour, let's say that for the getting pod a pod label will be include no? 

```
https://192.168.36.11:6443/api/v1/namespaces/kube-system/pods/cilium-25g7k
```

Also, I would not use response code and use success=true|false